src/integration/masternode/application/services/masternode.service.ts

```diff
@@ -143,7 +180,17 @@ export class MasternodeService {
     if (!masternode) throw new NotFoundException('Masternode not found');
     if (masternode.creationHash) throw new ConflictException('Masternode already created');
 
-    masternode.state = MasternodeState.CREATED;
+    try {
+      masternode.creationHash = await this.whaleService.broadcast(dto.signedTx);
```

Comment:
This may throw an error on waiting for transaction. I would do the wait in a second step (after DB Update).